libsolidity/analysis/ViewPureChecker.cpp

```diff
@@ -237,6 +239,15 @@ void ViewPureChecker::reportMutability(StateMutability _mutability, SourceLocati
 				", but this expression (potentially) modifies the state and thus "
 				"requires non-payable (the default) or payable."
 			);
+		else if (_mutability == StateMutability::Payable)
+		{
+			if (m_modifierInvocation)
+				m_errorReporter.warning(
```

Comment:
Shouldn't this be elevated from warning to error?

Reply:
That's a good question. Using ``msg.value`` in a non-payable function also results in a warning: ``"msg.value" used in non-payable function. Do you want to add the "payable" modifier to this function?"``
I'd either elevate both from warning to error or none of them. As far as I understand, this is a security feature that doesn't introduce undefined behaviour (please correct me if I wrong here :wink:), so we could also just keep the warning.

Reply:
Ah ok, if the other case is a warning, this PR shouldn't change that, but we should discuss that separately and make a decision.

Reply:
Tracked in #4601.